include/deal.II/particles/particle_accessor.h

```diff
@@ -875,7 +885,8 @@ namespace Particles
   ParticleAccessor<dim, spacedim>::operator==(
     const ParticleAccessor<dim, spacedim> &other) const
   {
-    return (property_pool == other.property_pool) && (cell == other.cell) &&
+    return (property_pool == other.property_pool) &&
+           (particles_in_cell == other.particles_in_cell) &&
            (particle_index_within_cell == other.particle_index_within_cell);
```

Comment:
I reworked the particle iterator so that we only need the `particles_in_cell` (i.e., `std::list<ParticleInCell>::iterator`), while completely skipping the pointers to particles. Regarding the comparisons: I think we need the comparison of iterators, at least for now, because we need to be able to compare with the `past-by-one` iterator, where we cannot access the handle. So I think this is the right way to go, and indeed if anything we are more restrictive if we consider the iterators to match as well.